content/tutorials/reactor-event-cloudflare.textile

```diff
@@ -0,0 +1,256 @@
+
+h2. Building the Game
+
+To get you up and running quickly, we've implemented this 2D game and made it available via this "tutorial's repository":https://github.com/ably/tutorials/tree/reactor-event-cloudflare-javascript. You can clone this repository and install its dependencies as per its readme.
+
+h2. Step 2: Creating your function
+
+Cloudflare workers is a Function-as-a-Service (FaaS) platform operated by Cloudflare that lets you run pieces of code when they are triggered. They're comparable to AWS Lambda, or Azure Functions.
+
+Cloudflare have an npm tool called "Wrangler":https://developers.cloudflare.com/workers/quickstart/ that they use to create their Workers. You can install this using:
+
+```[bash]
+npm install -g @cloudflare/wrangler
+```
+
+and then you can create new Workers by using the command `wrangler generate <my-worker-name>`.
+
+If you want to learn more about Wrangler and Cloudflare Workers, you can "read their documentation":https://developers.cloudflare.com/workers/ on the Cloudflare developer site.
+
+For this example, we've already done this for you and you'll find a Worker in the *./cloudflare-worker/* directory in your cloned repository. It looks a little like this:
+
+```[js]
+addEventListener('fetch', event => {
+    event.respondWith(handleRequest(event.request))
+})
+
+async function handleRequest(request) {
+    try {
+        const body = await request.text();   // Get the body from the response
+        const data = JSON.parse(body);       // Parse body to JavaScript object
+
+        if (!data) {
+            return jsonResponse(null, 'Request Body cannot be empty!');
+        }
+
+        /* Ably specific code here in the sample */
+
+        return new Response ({ text: "I'm ok!" });
+
+    } catch (error) {
+        return new Response(error, { status: 500 });
+    }
+}
+
+```
+
+h2. How the game works
+
+The game is a React app, created using "Create-React-App":https://github.com/facebook/create-react-app, that uses the react-router-dom npm package for routing.
+It's made up of two top level React components, one for the game and the other for an admin dashboard.
+
+The player uses the arrow keys on their keyboard to drive a post van around five pre-set maps (which can be seen in the [MAPS DIRECTORY]). The objective of the game is to deliver a parcel to a postbox on each level. Every time a parcel is delivered, we're publishing a message to an Ably channel and a new postbox is shown to the user.
+
+The Ably Reactor will listen for these publish events and call our Cloudflare Worker with the body of the published message. Once our Cloudflare Worker receives the message from the Ably Reactor, it will publish its own message to the Ably Channel. The dashboard is subscribed to these messages and will display the numnber of moves it took for the player to complete each level. In a real world scenario this might be the number of deliveries left on our driver's shift.
```

Comment:
I see a mix of our, your and the.  This is creating inconsistencies.  Given this is a tutorial, and they are following steps, I would assume if "You can install this using:", then they the Cloudflare work will be theirs.